### PR TITLE
OCPEDGE-1424: feat(controller): Create KSANNode objects

### PR DESCRIPTION
**Pull request by qJkee**
Create KSANNode objects based on user provided CR

### CODE REVIEW

File: api/v1alpha1/ksanstorage_types.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ type KSANStorageSpec struct {
 	KubesanParams map[string]string `json:"kubesanParams"`
 
 	// +kubebuilder:validation:Optional
-	Affinity v1.Affinity `json:"affinity,omitempty"`
+	NodeSelector *v1.NodeSelector `json:"affinity,omitempty"`
```
Comment (suleymanakbas91) [inline api/v1alpha1/ksanstorage_types.go:51]:
```suggestion
	NodeSelector *v1.NodeSelector `json:"nodeSelector,omitempty"`
```

File: internal/controller/node_selector.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	for _, node := range nodes.Items {
+		ok, err := corev1helper.MatchNodeSelectorTerms(&node, nodeSelector)
+		if err != nil {
+			return nil, err
+		}
+		if ok {
+			res = append(res, node)
+		}
+	}
+	return res, nil
```
Comment (suleymanakbas91) [inline internal/controller/node_selector.go:30]: What about taints and tolerations? Are we going to support them at a later time?

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: internal/controller/ksanstorage_controller.go [hunk 3]
```diff
@@ -27,6 +27,8 @@ import (
 	ksanv1alpha1 "openshift/ksan-operator/api/v1alpha1"
 )
 
+const operatorNamespace = "openshift-ksan-operator"
```
Comment (jeff-roche) [inline internal/controller/ksanstorage_controller.go:30]: We shouldn't hardcode this, we should use the environment to pull what namespace the operator is running in. LVMS does it like this:
https://github.com/openshift/lvm-operator/blob/main/internal/cluster/namespace.go

**Review by jeff-roche — CHANGES_REQUESTED** [review]

**Pull request merged**